doc/dev/tma.md

```diff
@@ -0,0 +1,71 @@
+# TMA support in nvFuser
+
+## Introduction
+
+TMA is a hardware feature that allows the GPU to load a tile from a tensor of up to 5D.
+The tile can be dense, or strided, as shown in Figure 1 below:
+
+![Figure 1: TMA dense and strided tile](tma/dense-and-strided-tile.svg)
+
+When using phrases like "tile size", it is important to be clear what we are referring to.
+Here we use two separate words *box* and *tile* to refer to different things.
+
+**Definition 1**: In an N-dimensional tensor, a *box* of size `(s1, s2, ..., sN)` at
+`(x1, x2, ..., xN)` refers to all the `s1*s2*...*sN` items at `(x1, x2, ..., xN)`,
+`(x1, x2, ..., xN + 1)`, ..., `(x1, x2, ..., xN + sN - 1)`, ..., `(x1 + s1 - 1, x2 + s2 - 1, ..., xN + sN - 1)`.
+
+**Definition 2**: In an N-dimensional tensor, a *tile* of size `(s1, s2, ..., sN)` and stride
+`(r1, r2, ..., rN)` at `(x1, x2, ..., xN)` refers to all the `s1*s2*...*sN` items at
+`(x1, x2, ..., xN)`, `(x1, x2, ..., xN + rN)`, ..., `(x1, x2, ..., xN + rN * (sN - 1))`,
+..., `(x1 + r1 * (s1 - 1), x2 + r2 * (s2 - 1), ..., xN + rN * (sN - 1))`.
+
+In Figure 1, we have box size `(6, 4)` for both diagram.
+For the diagram on the left, we have tile size `(6, 4)` and stride `(1, 1)`.
+For the diagram on the right, we have tile size `(6, 2)`, and stride `(1, 3)`.
+
+## Schedule
+
+In order to use TMA, we need to tell the hardware what is the dimensionality of our tensor.
+Most naively, we can make this dimensionality the size of the allocation domain.
+This naive mental model does provide us an easy way to reason about and start using TMA,
+but unfortunately it is not flexible enough and is not consistent with how we think about scheduling.
+For example, if we want to schedule a fusion containing only pure pointwise (i.e. no broadcasting) operations and all input and output tensors are contiguous,
+regardless of the actual dimensionality of the input and output tensors, in our mental model, we always
+consider this problem as a 1D problem by viewing all tensors as flattened 1D tensor.
+For this case, ideally, we should be using 1D TMA, instead of using the actual dimensionality of the tensor.
```

Comment:
Side question: Is there any perf difference by the dimensionality?

Reply:
I don't know. Never benchmarked.